### PR TITLE
Add cookie attribute

### LINKED ISSUE #4227: Cookies should set `SameSite=None` & Secure
**Is your feature request related to a problem? Please describe.**
Chrome is changing their policy on setting cookies.

The below is a warning message we see in the chrome console, future releases of chrome will stop delivering/settings cookies.

```
A cookie associated with a cross-site resource at http://bbc.co.uk/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.
```

image

Chrome Policies:
https://www.chromestatus.com/feature/5088147346030592
https://www.chromestatus.com/feature/5633521622188032

**Describe the solution you'd like**
When setting a cookie on a cross site domain i.e. bbc.com > bbc.co.uk ensure that we add the `SameSite=None` and `Secure` attributes.

**Testing notes**

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by thekp**
Resolves #4227
Duplicate of: #4296 

**Overall change:** _Updated our consentBanner logic to add additional attributes: `sameSite=None` and `secure=true`_

**Code changes:**

- _Updated the setCookie function to have additional attributes_
- _Update unit tests and snapshots_
- _Updated cypress tests for cookie banner to not run on local env because "secure" attribute will prevent cookies from being set on HTTP (which our local env is...)_
- _Updated HeadContainer test to use `news` service data_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by thekp** [comment]
repost: https://github.com/bbc/simorgh/pull/4296#issuecomment-545109073

Please ensure this PR is tested to ensure no cookie banner regressions ACROSS DEVICES. Espesh lower end and older devices we support

This includes testing that the banner doesnt continually reappear etc

File: cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js [hunk 1]
```diff
@@ -73,93 +73,98 @@ const visitPage = (service, pageType) => {
...
-          });
+              visitPage(service, pageType);
 
-          it('should have a privacy banner that disappears once accepted but a cookie banner that is rejected', () => {
-            cy.clearCookies();
-            visitPage(service, pageType);
+              assertCookieValues({
+                ckns_explicit: '1',
+                ckns_privacy: 'july2019',
+                ckns_policy: '000',
```
Comment (oluoluoxenfree) [inline cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js:140]: What do these assertions mean?

Comment (thekp) [reply cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js:140]: it uses this method: https://github.com/bbc/simorgh/blob/71b50d1c65d33bc4baebbb8e7066c2e3d26067ca/cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js#L64-L68

it checks the cookie (ckns_explicit, ckns_privacy, ckns_policy) has the corresponding value assigned to it.

in this case `ckns_explicit ` has been accepted so the value is `1` and cookie banner was rejected so the value is `000`.

File: cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js [hunk 2]
```diff
@@ -73,93 +73,98 @@ const visitPage = (service, pageType) => {
...
-            getPrivacyBanner(service, variant).should('be.visible');
-            getCookieBanner(service, variant).should('not.be.visible');
-
-            assertCookieValues({
-              ckns_privacy: 'july2019',
-              ckns_policy: '000',
+// Note: This test is disabled on local env because the "secure=true" attribute was introduced
+// to our setCookie functionality in this PR: https://github.com/bbc/simorgh/pull/4296
+// Our localhost is currently insecure and the "secure" attribute will prevent cookies being set on HTTP.
+if (Cypress.env('APP_ENV') !== 'local') {
```
Comment (thekp) [inline cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js:79]: FYI, this is my change to the file ^ (to not run the cypress test on local)

File: cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js (hunk 1 above)
Comment (thekp) [reply cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js:140]: (These tests already existed, the only change I made is here: https://github.com/bbc/simorgh/pull/4391/files#diff-b516ab6e536a3fd977362f8719c08543R76-R79)

github made it looked weird 🤕

**Review by oluoluoxenfree — APPROVED** [review]

**Comment by mykeels** [comment]
> Updated cypress tests for cookie banner to not run on local env because "secure" attribute will prevent cookies from being set on HTTP (which our local env is...)

Are we working on making our local env https? Is CI run with http too?

**Comment by thekp** [comment]
@mykeels 
> Are we working on making our local env https? Is CI run with http too?

I don't believe we have any issues to make local env https. And I believe CI will be http. @hindsc52 may have more context, if we have future plans on making our local env https.


edit: discussed on slack
edit: CI is run against test and live

**Comment by jamesdonoh** [comment]
CI runs against localhost and therefore uses http. CD (when we make a release) includes running tests against test and live, which use https.

**Comment by thekp** [comment]
I am waiting for a response from the Orbit team about this change. We want to ensure we have the same behaviour as their cookie attributes.

**Pull request closed without merging**